### DISCUSSION TITLE
decomposition failed

### DISCUSSION THREAD

**Discussion by AimZhang** [peridigm/peridigm#280] [General]
Dear sir or mamdame,

I am new beginner of Peridigm. I am trying to decomposite the "disk_impact.g" obtained by Trelis. I used the command /mnt/e/Peridigm/trilinos/bin/decomp -p 4  disk_impact.g, but I got the error. Below is a detailed description of the error. Could you please give me some advice or guidelines? Thank you very much!

Error:
Executing:
   /mnt/e/Peridigm/trilinos/bin/nem_spread  disk_impact.g.pex

nem_spread version 7.01 (2021/03/19)
Using 4 byte integers and 8 byte floating point values.
Reading the command file, disk_impact.g.pex

Exodus file (disk_impact.g)
        Title of file: 'cubit(impact/cu-cu mesh/cu-cu mesh 3/cu-cu mesh 4/disk_impact.g): 11/09/2022: 14'
        Dimensionality of problem =              3
        Number of nodes           =      5,356,030
        Number of elements        =      5,377,085
        Number of element blocks  =              2
        Number of node sets       =              1
        Number of side sets       =              0

Start to read in and distribute the load balance info
EXODUS II load-balance file: disk_impact.g.nem

Finished distributing load balance info

Load load balance information time: 0.30715169999999997 (sec.)

Load exoII restart param info to each proc.

Load restart parameters time: 0.0003359000000000001 (sec.)

Load exoII mesh info to each proc.

        Time to read element block IDs:            0.0001
        Time to read node set IDs:                 0.0001
        Time to extract element block information: 0.2729
        Time to read nodal coordinates:            0.4610
        Time to read element blocks:               0.8259
        Time to read node sets:                    0.0765
/mnt/e/Peridigm/trilinos/bin/decomp: line 188:   728 Aborted                 (core dumped) $NEM_SPREAD $use64 $pexsh

ERROR:******************************************************************
ERROR:
ERROR    During nem_spread execution. Check error output above and rerun
ERROR:
ERROR:******************************************************************

**Comment by ppandit95** [comment]
Hi AimZhang

This error pertains to Trilinos Library so I would help out if you post it on the discussion forum of Trilinos as I guess the trilinos installation is not properly done .

Have you tried decomposing other geometries specified in example folder.

Hope this helps.

Regards
Pushkar

**Reply by AimZhang** [reply]
Dear Pushkar,

I have tried to decompose the same geometry into a different number of files (/mnt/e/Peridigm/trilinos/bin/decomp -p 60 disk_impact.g **not** /mnt/e/Peridigm/trilinos/bin/decomp -p 4 disk_impact.g ), it can work well. It's strange.

**Reply by ppandit95** [reply]
This means that the Trilinos is installed correctly then I guess the geometry created in .g is not fit to be decomposed into 4 parts and in case if it is not a strict constraint to decompose your problem in 4 parts then I would suggest you to go with larger no. of decompositions than 4 .

Hope this Helps

Regards
Pushkar

**Comment by AimZhang** [comment]
Dear Pushkar,
Thank you for your suggestions. If I use the supercomputer to do calculations, I can decompose it into larger parts. But the problem is my own computer only has 6 cores. That's why I want to decompose it into 4 or 6 parts.

Best regards,

**Reply by ppandit95** [reply]
Maybe you can try with 2 parts....
